src/auth.rs

```diff
@@ -0,0 +1,133 @@
+#![deny(missing_docs)]
+//! Ways to authenticate with the Webex API
```

Comment:
Being a bit generic here even though we only have `DeviceAuthenticator` for now. Assuming in the feature we can have multiple flows and maybe a trait for authentication.